dspace-api/src/main/java/org/dspace/versioning/service/VersioningService.java

```diff
@@ -53,4 +53,6 @@
     Version getVersion(Context c, Item item) throws SQLException;
 
     Version createNewVersion(Context context, VersionHistory history, Item item, String summary, Date date, int versionNumber);
+
+    void removeVersionFromItem(Context context, Item item);
```

Comment:
This should probably throw exceptions, just like the other `remove*` methods in `VersioningService`.  So, I'm surprised to see it *not* have `throws SQLException, AuthorizationException`.  This results in you having to catch those errors and transform them into `RuntimeException` (which isn't the best behavior for an API).